lib/aws/client.ex

```diff
@@ -13,7 +13,7 @@ defmodule AWS.Client do
             port: "443",
             http_client: {HTTPoison, []},
             json_module: {Poison, [encode: :encode!, decode: :decode!]},
```

Comment:
This should look like this now:

    {AWS.JSON, []}

And you can implement both functions that call poison. :) Same for XML and HTTPClient. They should all call their behaviour and the behaviour now includes the default implementation!